src/commands/command.ts

```diff
@@ -24,4 +25,15 @@ export interface Command {
   description: string;
   args: ArgDescriptor[];
   run(options: CommandOptions, config: ProjectConfig): Promise<any>;
-}
\ No newline at end of file
+  extraUsageGroups?(config: ProjectConfig): UsageGroup[];
+}
+
+/**
+ * Running a command may reject with a CommandFailure to indicate that there
+ * was an error but it has already been reported to the user, and the CLI
+ * should exit with the given exit code.
+ */
+export class CommandFailure {
```

Comment:
Great first step! In the future I'd like to see this extend the error type have a message / stack trace / more info that we could output when verbose is true.